client/src/main/java/com/paypal/selion/platform/grid/browsercapabilities/DefaultCapabilitiesBuilder.java

```diff
@@ -49,9 +49,14 @@ public DesiredCapabilities getDefaultCapabilities() {
 
         capability.setCapability(CapabilityType.ACCEPT_SSL_CERTS, true);
 
-        // refer here : http://code.google.com/p/selenium/wiki/DesiredCapabilities#Read-write_capabilities
+        // refer here: http://code.google.com/p/selenium/wiki/DesiredCapabilities#Read-write_capabilities
         // for understanding the relevance of this capability.
-        capability.setCapability(CapabilityType.UNEXPECTED_ALERT_BEHAVIOUR, "ignore");
+        // marionette does not like this capability. So, let's not set it in that
+        // instance.
+        if (!Grid.getWebTestSession().getBrowser().contains("firefox")
+                && getLocalConfigProperty(ConfigProperty.SELENIUM_USE_GECKODRIVER).equals("true")) {
+            capability.setCapability(CapabilityType.UNEXPECTED_ALERT_BEHAVIOUR, "ignore");
+        }
```

Comment:
ideally, yes. However, this method in `DefaultCapabilitiesBuilder` happens after `FireFoxCapabilitiesBuilder`.  I'll see if I can devise a more elegant approach